incubator/hnc/api/v1alpha1/hnc_config.go

```diff
@@ -88,6 +95,9 @@ type HNCConfigurationSpec struct {
 type HNCConfigurationStatus struct {
 	// Types indicates the observed synchronization states of kinds, if any.
 	Types []TypeSynchronizationStatus `json:"types,omitempty"`
+
+	// Conditions describes the errors, if any.
+	Conditions []HNCConfigurationCondition `json:"conditions,omitempty"`
```

Comment:
I was considering adding another field in `HNCConfigurationCondition` struct to indicate types when I did the implementation, e.g.,

```go
  type HNCConfigurationCondition struct {
    Code HNCConfigurationCode `json:"code"`
    Msg string `json:"msg,omitempty"`
    Affects []AffectedType `json:"affects, omitempty"`
 }

type AffectedType struct {
	APIVersion string `json:"apiVersion,omitempty"`
	Kind string `json:"kind,omitempty"`
}
```

But I didn't end up doing that because:
- Right now we only have two error conditions. The one that are associated with types (`objectReconcilerCreationFailed`) already has the type information in the Msg field. Adding the Affects field does not seem to provide any additional information.
- Also the other condition (`critSingletonNameInvalid`) is not related to any type and will have an empty `Affects` field. This might cause some confusion because when the field is empty we might not be able to tell if it impacts all types or if we miss the type information.

I am thinking about adding the field later if we end up having many error codes associated with specific types or if we find it is not enough to provide type information in the error message. Feel free to let me know if we want to include it in this PR or if there are any other approaches to associate a condition with types (I agreed with what you said that putting `Condition` in `Type` might be troublesome because some errors are not tied to any types).

Reply:
No, I think you're right. I've been focused on trying to make everything machine-readable, but in this case I think it's fine to just mention the bad Kind in the human-readable error message. Maybe just make sure the documentation to the codes clearly state that `msg` will specify whatever information is needed.

Reply:
Cool! Updated the documentation of Msg to say that it will include type information if the condition is for specific types.